webapp/src/main/java/com/box/l10n/mojito/service/ai/translation/AITranslateCronJob.java

```diff
@@ -292,6 +303,14 @@ private boolean isExpired(TmTextUnitPendingMT pendingMT) {
                 .minus(aiTranslationConfiguration.getExpiryDuration()));
   }
 
+  private boolean isUsed(TMTextUnit textUnit) {
+    TextUnitSearcherParameters params = new TextUnitSearcherParameters();
+    params.setTmTextUnitIds(List.of(textUnit.getId()));
+    params.setUsedFilter(UsedFilter.USED);
+    List<TextUnitDTO> result = textUnitSearcher.search(params);
```

Comment:
This can be quite an expensive query to run on one TU at a time

Would it be possible to run this query once at a batch level and pass down the info or filter earlier?